bluemira/radiation_transport/radiation_profile.py

```diff
@@ -0,0 +1,1752 @@
+        ne_mp,
+        core=False,
+        n_rad_in=None,
+        n_rad_out=None,
+        rad_i=None,
+        rec_i=None,
+        n_tar=None,
+        x_point_rad=False,
+    ):
+        """
+        Along a single flux tube, it assignes different density values.
+
+        Parameters
+        ----------
+        flux_tube: loop
+            flux tube geometry
+        ne_mp: float
+            electron density at the midplane
+        core: boolean
+            if True, n is constant along the flux tube. If false,it varies
+        n_rad_in: float
+            density value at the entrance of the radiation region
+        n_rad_out: float
+            density value at the exit of the radiation region
+        rad_i: np.array
+            indeces of points, belonging to the flux tube, which fall
+            into the radiation region
+        rec_i: np.array
+            indeces of points, belonging to the flux tube, which fall
+            into the recycling region
+        n_tar: float
+            electron density at the target
+        x_point_rad: boolean
+            if True, it assumes there is no radiation at all in the recycling region
+
+        Returns
+        -------
+        ne: np.array
+            poloidal distribution of electron desnity
+        """
+        # initialising ne with same values all along the flux tube
+        ne = np.array([ne_mp] * len(flux_tube))
+        if core is True:
+            return ne
+
+        if rad_i is not None and len(rad_i) == 1:
+            ne[rad_i] = n_rad_in
+        elif rad_i is not None and len(rad_i) > 1:
+            ne[rad_i] = self.exponential_decay(n_rad_out, n_rad_in, len(rad_i))
+
+        if rec_i is not None and x_point_rad:
+            ne[rec_i] = self.gaussian_decay(n_rad_out, 0.1, len(rec_i))
+        elif rec_i is not None and x_point_rad is False:
+            ne[rec_i] = self.gaussian_decay(n_rad_out, n_tar, len(rec_i))
+
+        return ne
+
+    def plot_1d_profile(self, rho, rad_power, ax=None):
+        """
+        1D plot of the radation power distribution along the midplane.
+        if rad_power.ndim > 1 the plot shows as many line as the number
+        of impurities, plus the total radiated power
+
+        Parameters
+        ----------
+        rho: np.array
+            dimensionless radius. Values between 0 and 1 for the plasma core.
+            Values > 1 for the scrape-off layer
+        rad_power: np.array
+            radiated power at each mid-plane location corresponding to rho.
+        """
+        if ax is None:
+            fig, ax = plt.subplots()
+            plt.xlabel("rho")
+            plt.ylabel("MW/m^3")
+
+        if len(rad_power) == 1:
+            ax.plot(rho, rad_power)
+        else:
+            [ax.plot(rho, rad_part) for rad_part in rad_power]
+            rad_tot = np.sum(np.array(rad_power, dtype=object), axis=0).tolist()
+            ax.plot(rho, rad_tot)
+
+        return ax
+
+
+class TwoPointModelTools(Radiation):
```

Comment:
We don't use classes as function namespaces. All of these methods look like functions and shouldn't need to initialise a namespace with Parameters in order to get this behaviour.